app/services/immediate_email_generation_service/batch.rb

```diff
@@ -0,0 +1,92 @@
+class ImmediateEmailGenerationService
+  class Batch
+    def initialize(content, subscription_ids_by_subscriber_id)
+      @content = content
+      @subscription_ids_by_subscriber_id = subscription_ids_by_subscriber_id
+    end
+
+    def generate_emails
+      email_ids = []
+      return email_ids unless email_parameters.any?
+
+      ActiveRecord::Base.transaction do
+        email_ids = ContentChangeEmailBuilder.call(email_parameters) if content_change
+        email_ids = MessageEmailBuilder.call(email_parameters) if message
+
+        records = subscription_content_records(email_ids)
+        SubscriptionContent.populate_for_content(content, records)
+      end
+
+      email_ids
+    end
+
+  private
+
+    attr_reader :subscription_ids_by_subscriber_id, :content
+
+    def email_parameters
+      @email_parameters ||= begin
+        subscriptions_to_fulfill_by_subscriber.map do |(subscriber, subscriptions)|
+          {
+            address: subscriber.address,
+            content_change: content_change,
+            message: message,
+            subscriptions: subscriptions,
+            subscriber_id: subscriber.id,
+          }.compact
```

Comment:
```suggestion
          }
```

This looks like it was left over from something.

Reply:
We still need the compact as it removes the content_change option when dealing with a message and vice-versa.